core/foundation/src/StringUtils.cxx

```diff
@@ -37,4 +41,57 @@ std::vector<std::string> Split(std::string_view str, std::string_view delims, bo
    return out;
 }
 
+/**
+ * \brief Convert (round) a value and its uncertainty to string using one or two significant digits of the error
+ * \param error the error. If the error is negative or zero, only the value is returned with no specific rounding applied, using std::to_string
+ * \param cutoff should lay between 0 and 9. If first significant digit of error starts with value <= cutoff, use two significant digits instead of two for
+ * rounding. Set this value to zero to always use a single digit; set this value to 9 to always use two digits.
+ * \param delim delimiter between value and error printed into returned string, leave default for using ROOT's latex mode
+ * \return a string with printed rounded value and error separated by "+/-" in ROOT latex mode
+ * \note The return format is `A+-B` using ios::fixed with the proper precision;
+ * for very large or very small values of the error, the format is changed from `A+-B` to (A'+-B')*1eX, with X being multiple of 3, respecting the corresponding precision.
+ * \see https://www.bipm.org/en/doi/10.59161/jcgm100-2008e, https://physics.nist.gov/cuu/Uncertainty/
+ */
+std::string Round(double value, double error, unsigned int cutoff, std::string_view delim)
+{
+   if(error <= 0.)
+   {
+      return std::to_string(value);
+   }
+
+   int nexp10 = std::floor(std::log10(error));
+   const auto scale = std::pow(10., nexp10);
+   const auto first_digit = static_cast<unsigned int>(error / scale);
+   assert (first_digit > 0 && first_digit < 10);
+   if (first_digit <= cutoff) {
+      const double rerror = error * std::pow(10., -1. * nexp10);
+      if (static_cast<unsigned int>(std::round(rerror * 10) / 10) <= cutoff)
+         nexp10--;
+   } else if (cutoff == 0 && first_digit == 9) {
+      const double rerror = std::round(error * std::pow(10., -1. * nexp10));
+      const int rnexp10 = std::floor(std::log10(rerror));
+      const auto rscale = std::pow(10., rnexp10);
+      const auto rfirst_digit = static_cast<int>(rerror / rscale);
+      if (rfirst_digit == 1)
+         nexp10++;
+   }
+
+   std::stringstream sv, se;
+   sv.setf(std::ios::fixed);
+   se.setf(std::ios::fixed);
+   const int maxExpo = error <= 1e-3 ? static_cast<int>(std::floor(std::log10(error) / 3)) * 3 : static_cast<int>(std::log10(error) / 3) * 3;
+   if(nexp10 - maxExpo < 0) {
+      sv.precision(-nexp10 + maxExpo);
+      se.precision(-nexp10 + maxExpo);
+   }
+   else {
+      sv.precision(0);
+      se.precision(0);
+   }
+   sv << std::round(value * std::pow(10., -nexp10))/std::pow(10., -nexp10 + maxExpo);
+   se << std::round(error * std::pow(10., -nexp10))/std::pow(10., -nexp10 + maxExpo);
+
+   return (maxExpo != 0 ? "(" : "") + sv.str() + std::string(delim) + se.str() + (maxExpo != 0 ? ")*1e" + std::to_string(maxExpo) : "") ;
```

Comment:
Since we're already using `stringstream`, I think it's worth using also for this long concatenation.

Reply:
I've reordered for better legibility, could you take a look if it's ok like now? I thought it was cleaner for readability to go back to 'strings' for the result.